### ISSUE TITLE
[Source Request] AnimeTosho follow-up

### ISSUE THREAD

**Issue by Nuklear92** [Goldenfreddy0703/Otaku#132]
I originally suggested adding this source aside from Nyaa which is currently the only torrent source available.

It was rejected due to the confirmation that they mostly gather their torrents from Nyaa. However, lately I've noticed that Nyaa scraping times had been very finicky or their mainly their website servers having ups and downtimes. 

However, this is where https://animetosho.org comes in place to take place for when Nyaa scrapping gets a bit of hit or miss. This website in particular seems to be way more responsive than Nyaa lately, and I've been having it as my go to when searching anime content and so far it has been way faster and more reliable in search times and not hanging around. Which I certainly believe it would translate the same way if it was added as a secondary torrent source for scraping Debrid which if Nyaa can't make it, AnimeTosho will back it up and hopefully still end up pulling all of the Nyaa results + who knows if something extra appears maybe if they somehow get stuff from other places as well in the background. 

In conclusion, this won't definitely hurt getting this source added and will add more accuracy in the first scraping which there's often times when I scrape something the first scrape Nyaa gets timed out and I have to back out and then re-scrape and we could hopefully eliminate that hassle if we had both sources to complement and take over each other when one isn't performing reliably in a given time.

**Comment by Goldenfreddy0703** [comment]
Okay I do see where your coming from and I do agree that scrapping times have been very finicky these past few months due to how nyaa works with Otaku. Nyaa is a bit of a mess with Otaku and we have to fix it very soon. Luckily we do have new contributors that can help us with nyaa. @Pahiro last time did a fantastic job with context menu and I think in the future, he will be able to help with Nyaa.

One main problem I have noticed is that season or part number is not getting recorded on nyaa which causes nyaa to pull a episode from a random season. In the future we will have to fix that soon, That's also why I made those prioritization settings in the first place sense nyaa has scrapping problems. 

Luckily we will have a fix with nyaa very soon. After this fix, if scrapping is still kinda finicky then we will consider on thinking if adding animetosho in the future.

**Comment by Nuklear92** [comment]
Fair enough, I agree with Nyaa scrapping random episodes on a lot of anime I watch and returning seasons. Glad, this fix is on the radar and hopefully by making it more accurate to scraping content perhaps stuff that usually doesn't get in the results even if it's cached in the debrid service aftwards start populating in the results. Because, that's another issue Nyaa scraping seems to have and that it only seems to pick up about 60-80% of links and not all of them regardless of them being cached. I've come across a lot of older anime seasons which have a good couple of Blu-ray season packs when you visit the Nyaa website. However when you scrape that season in otaku it only picks up some season packs and episodes but not all of them. Sometimes I have to manually cache the particular season pack I want to watch and make otaku check my debrid cloud in order for it to populate in the results.

Sorry for the long comment, overall I'm glad that this will get a treatment soon so we can get more accurate results. 👍

**Comment by Goldenfreddy0703** [comment]
Hey so very good news @Nuklear92 , we have a fixed version of Nyaa and we have added Anime Tosho Scrapper. Soon we will make a release either today or tomorrow or maybe the weekend.  We still have some other settings to add soon but everything looks like its going great.

**Comment by Nuklear92** [comment]
> Hey so very good news @Nuklear92 , we have a fixed version of Nyaa and we have added Anime Tosho Scrapper. Soon we will make a release either today or tomorrow or maybe the weekend.  We still have some other settings to add soon but everything looks like its going great. 
> 
> 

Whoa! Great news, can't wait. 😃 thanks for all the hard work you had been putting into this addon to make it the best possible one stop for all anime stuff! 💙

**Comment by Goldenfreddy0703** [comment]
Hey @Nuklear92 Good news, we just made a release just now, feel free to update it. Huge thanks to @joezito129 for making this all happen. Without him, we would not have a nyaa fix at all.

**Comment by Nuklear92** [comment]
Hey @Goldenfreddy0703 Just wanted to say thanks a lot for the new update! The scrapping now is super accurate and the AnimeTosho was a great addition! I can already notice some links from it populating in the results alongside Nyaa. Not being duplicates but mainly links that Nyaa alone didn't had. I believe AnimeTosho also has other sources where it picks their torrents aside from Nyaa being one of them. And you can tell there's now more links and season packs populating that they didn't showed before you fixed the scraping and accuracy.

This is by far one of the most exciting and great releases you have done in a while. Not putting any of the other releases in shame, but this one was a must have and greatly has improved the results from every anime I've searched and scrape so far. 

Again, thanks a lot for all the work you have put into this addon! 💙

**Comment by Goldenfreddy0703** [comment]
Hey thanks but this release especially is all thanks to @joezito129 without him, nyaa would still be broken and we would not have any clue to fix it. I think this one is a really great release cause nyaa has been broken for years and it's always been a mess. With this release tho, it's been amazing. In the future we may do a whole watchlist rewrite but it will take alot of time tho but hopefully we can get that done.

**Comment by Nuklear92** [comment]
Yeah, I forgot to mention him since you tagged his name when you replied me about Nyaa getting fixed. @joezito129 really appreciate the work you did on this release! Glad to see contributors like you making a great addon, even better! 💙
